packages/capacitor-plugin/src/definitions.ts

```diff
@@ -0,0 +1,565 @@
+   *
+   * @since 1.0.0
+   */
+  directory?: Directory;
+}
+
+export interface GetUriOptions {
+  /**
+   * The path of the file to get the URI for
+   *
+   * @since 1.0.0
+   */
+  path: string;
+
+  /**
+   * The `Directory` to get the file under
+   *
+   * @since 1.0.0
+   */
+  directory: Directory;
+}
+
+export interface StatOptions {
+  /**
+   * The path of the file to get data about
+   *
+   * @since 1.0.0
+   */
+  path: string;
+
+  /**
+   * The `Directory` to get the file under
+   *
+   * @since 1.0.0
+   */
+  directory?: Directory;
+}
+
+export interface CopyOptions {
+  /**
+   * The existing file or directory
+   *
+   * @since 1.0.0
+   */
+  from: string;
+
+  /**
+   * The destination file or directory
+   *
+   * @since 1.0.0
+   */
+  to: string;
+
+  /**
+   * The `Directory` containing the existing file or directory
+   *
+   * @since 1.0.0
+   */
+  directory?: Directory;
+
+  /**
+   * The `Directory` containing the destination file or directory. If not supplied will use the 'directory'
+   * parameter as the destination
+   *
+   * @since 1.0.0
+   */
+  toDirectory?: Directory;
+}
+
+export type RenameOptions = CopyOptions;
+
+export interface ReadFileResult {
+  /**
+   * The representation of the data contained in the file
+   *
+   * Note: Blob is only available on Web. On native, the data is returned as a string.
+   *
+   * @since 1.0.0
+   */
+  data: string | Blob;
+}
+
+export interface WriteFileResult {
+  /**
+   * The uri where the file was written into
+   *
+   * @since 1.0.0
+   */
+  uri: string;
+}
+
+export interface ReaddirResult {
+  /**
+   * List of files and directories inside the directory
+   *
+   * @since 1.0.0
+   */
+  files: FileInfo[];
+}
+
+export interface FileInfo {
+  /**
+   * Name of the file or directory.
+   */
+  name: string;
+  /**
+   * Type of the file.
+   *
+   * @since 1.0.0
+   */
+  type: 'directory' | 'file';
+
+  /**
+   * Size of the file in bytes.
+   *
+   * @since 1.0.0
+   */
+  size: number;
+
+  /**
+   * Time of creation in milliseconds.
+   *
+   * It's not available on Android 7 and older devices.
+   *
+   * @since 1.0.0
+   */
+  creationTime?: number;
+
+  /**
+   * Time of last modification in milliseconds.
+   *
+   * @since 1.0.0
+   */
+  modificationTime: number;
+
+  /**
+   * The uri of the file.
+   *
+   * @since 1.0.0
+   */
+  uri: string;
+}
+
+export interface GetUriResult {
+  /**
+   * The uri of the file
+   *
+   * @since 1.0.0
+   */
+  uri: string;
+}
+
+export type StatResult = FileInfo;
+export interface CopyResult {
+  /**
+   * The uri where the file was copied into
+   *
+   * @since 1.0.0
+   */
+  uri: string;
+}
+
+export interface FilesystemPlugin {
+  /**
+   * Read a file from disk
+   *
+   * @since 1.0.0
+   */
+  readFile(options: ReadFileOptions): Promise<ReadFileResult>;
+
+  /**
+   * Read a file from disk, in chunks
+   * Native only (not available in web)
+   * 
+   * @since 1.0.0
```

Comment:
This is only offered in the future plugin version, which I think will be

```suggestion
   * @since 7.1.0
```

Reply:
fixed